### ISSUE TITLE
Unicode alternates in PHP vs JS? (v2.1.0 via composer)

### ISSUE THREAD

**Issue by jensschulze** [Ranks/emojione#199]
These two emojis ⛔️ (with Variation Selector U+FE0F, pasted from Apple) and ⛔ (without Variation Selector, pasted from an email written in Gmail) are treated differently in the PHP `toShort()` method. The first one translates into its shortname `:no_entry:`, whilst the latter does not.
The Javascript `toShort()` method succeeds in both cases.
I’m not quite sure if I have missed something, but is it possible to translate them into `:no_entry:` with PHP only?

**Comment by mikebe11** [comment]
We'll have to investigate a little more but it looks like a PHP regexp might need to be revised to capture both variations.

**Comment by jensschulze** [comment]
As far as I can see the regex matches in both cases so the method Client::toShortCallback is called. But because of the explicit `':no_entry:' => "\xE2\x9B\x94\xEF\xB8\x8F",` in Ruleset.php (line 2270) only the emoji with the Variation Selector is converted. I changed this line to `':no_entry:' => "\xE2\x9B\x94",`, and in this case only the emoji without Variation Selector was converted.
If it is – unlike it’s done in the JS library (`':no_entry:':["26d4-fe0f","26d4"],`) – not possible to define alternates, I’d suggest to strip a trailing `\xEF\xB8\x8F`, maybe even everything in the range from `\xEF\xB8\x80` to `\xEF\xB8\x8F` first in Client::toShortCallback. I tried it and it works in this special case. If you think this is a practicable way of handling it (-> I am not a Unicode expert <-) I could run all your tests, and create a pull request.
